File: mvvm/src/models/muhoho.ts
export interface MuhohoI {
  get face(): string;
}

export class Muhoho implements MuhohoI {
  private static readonly LEFT_CHEEK = "( ";
  private static readonly RIGHT_CHEEK = ")";
  private static readonly DEFAULT_EYE = "^";
  private static readonly DEFAULT_MOUTH = "ω";

  constructor(
    private readonly eye: string = Muhoho.DEFAULT_EYE,
    private readonly mouth: string = Muhoho.DEFAULT_MOUTH
  ) {}

  get face(): string {
    return (
      Muhoho.LEFT_CHEEK + this.eye + this.mouth + this.eye + Muhoho.RIGHT_CHEEK
    );
  }
}

File: clean-architecture/domain/entities/muhoho.go
package entities

type Muhoho struct {
	LeftCheek  string
	RightCheek string
	Eye        string
	Mouth      string
}

func NewMuhoho(eye string, mouth string) *Muhoho {
	const LEFT_CHEEK = "("
	const RIGHT_CHEEK = ")"
	const DEFAULT_EYE = "^"
	const DEFAULT_MOUTH = "ω"

	if eye == "" {
		eye = DEFAULT_EYE
	}

	if mouth == "" {
		mouth = DEFAULT_MOUTH
	}

	return &Muhoho{
		LeftCheek:  LEFT_CHEEK,
		RightCheek: RIGHT_CHEEK,
		Eye:        eye,
		Mouth:      mouth,
	}
}
